### PR TITLE
[DOCS-8697] Add standard source metadata fields

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do? What is the motivation?

- Adds info on standard source metadata fields.
- Adds further reading section.
- Removes list of source in doc because it's already in the nav.

[DOCS-8697](https://datadoghq.atlassian.net/browse/DOCS-8697)

### Merge instructions

- [x] Please merge after reviewing

### Additional notes

[DOCS-8697]: https://datadoghq.atlassian.net/browse/DOCS-8697?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by neuronull — APPROVED** [review]
🎉

File: content/en/observability_pipelines/sources/_index.md [hunk 1]
```diff
@@ -2,9 +2,15 @@
 title: Sources
 disable_toc: false
 further_reading:
-- link: "logs/processing/pipelines"
+- link: "observability_pipelines/set_up_pipelines/"
```
Comment (alai97) [inline content/en/observability_pipelines/sources/_index.md:5]:
```suggestion
- link: "/observability_pipelines/set_up_pipelines/"
```

File: content/en/observability_pipelines/sources/_index.md [hunk 2]
```diff
@@ -2,9 +2,15 @@
...
   tag: "Documentation"
-  text: "Log processing pipelines"
+  text: "Set up pipelines"
+- link: "observability_pipelines/processors/"
```
Comment (alai97) [inline content/en/observability_pipelines/sources/_index.md:8]:
```suggestion
- link: "/observability_pipelines/processors/"
```

File: content/en/observability_pipelines/sources/_index.md [hunk 3]
```diff
@@ -2,9 +2,15 @@
...
+  tag: "Documentation"
+  text: "Processors for your pipelines"
+- link: "observability_pipelines/destinations/"
```
Comment (alai97) [inline content/en/observability_pipelines/sources/_index.md:11]:
```suggestion
- link: "/observability_pipelines/destinations/"
```

**Review by alai97 — APPROVED** [review]
Looks good, small link fixes.

**Pull request merged**